Server/integrations/saml-passport/SamlPassportAuthenticator.py

```diff
@@ -438,19 +485,48 @@ def validSignature(self, jwt):
         valid = False
 
         try:
+
             appConfiguration = AppConfiguration()
             appConfiguration.setWebKeysStorage(WebKeyStorage.KEYSTORE)
             appConfiguration.setKeyStoreFile(self.keyStoreFile)
             appConfiguration.setKeyStoreSecret(self.keyStorePassword)
             appConfiguration.setKeyRegenerationEnabled(False)
 
             cryptoProvider = CryptoProviderFactory.getCryptoProvider(appConfiguration)
-            valid = cryptoProvider.verifySignature(jwt.getSigningInput(), jwt.getEncodedSignature(), jwt.getHeader().getKeyId(),
-                                                        None, None, jwt.getHeader().getSignatureAlgorithm())
+            
+
+            alg_string = str(jwt.getHeader().getAlgorithm())
+            signature_string = str(jwt.getEncodedSignature())
+
+            if alg_string == "none" or alg_string == "None" or alg_string == "NoNe" or alg_string == "nONE" or alg_string == "NONE" or alg_string == "NonE" or alg_string == "nOnE":
+                # blocks none attack
+
+                print "WARNING: JWT Signature algorithm is none"
+                valid = False
+            
+            elif alg_string != "RS512":
+                # blocks anything that's not RS512
+            
+                print "WARNING: JWT Signature algorithm is NOT RS512"
+                valid = False
+            
+            elif signature_string == "" :
+                # blocks empty signature string
+                print "WARNING: JWT Signature not sent"
+                valid = False
+            
+            else:
+                #changed method from .getSignatureAlgorithm() to getAlgorithm()
+                # class that extends AbstractCryptoProvider
+                
+                valid = cryptoProvider.verifySignature(jwt.getSigningInput(), jwt.getEncodedSignature(), jwt.getHeader().getKeyId(),
```

Comment:
It's not clear how it should work since verifySignature method has SignatureAlgorithm in that method, see
https://github.com/GluuFederation/oxAuth/blob/master/Model/src/main/java/org/gluu/oxauth/model/crypto/OxAuthCryptoProvider.java#L263

It seems it should be `jwt.getHeader().getSignatureAlgorithm()`.
Did you test it?

Reply:
Hello @yuriyz ! Thx for checking. Yes I tested.

- `jwt.getHeader().getSignatureAlgorithm()` does not work on my gluu 4.1.1 (updated identity jar)
- `jwt.getHeader().getAlgorithm()` works and returns `alg` header field. 
- 

Just in case I gonna take some time right now to investigate a little further the non-working `getSignatureAlgorithm()` on my side, it was one of the first things I changed so I will refresh my memory and get back to you.

Reply:
This PR is to master. If you follow link I posted above you will see that in master verifySignature accepts SignatureAlgorithm. Thus it is not clear how it is going to work on master (4.2).

Reply:
Hello @yuriyz .
On my version without `customScript` as `PersonAuthentication` arg that [you can check in this commit](https://github.com/GluuFederation/oxAuth/commit/9f053092a3eeaa324444b796bb1ddb22fa42e506) i get:

```
2020-05-14 16:24:57,116 INFO  [qtp1590550415-62] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Exception: 
2020-05-14 16:24:57,117 INFO  [qtp1590550415-62] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) -  'org.gluu.oxauth.model.jwt.JwtHeader' object has no attribute 'getSignatureAlgorithm'
```

Reply:
> This PR is to master. If you follow link I posted above you will see that in master verifySignature accepts SignatureAlgorithm. Thus it is not clear how it is going to work on master (4.2).

Sorry conversation wasn't updated...  checking.

Reply:
Here is diff of `JwtHeader` between 4.1 and master. So yes, in 4.1 it's `getAlgorithm()` but in master (4.2) it's `getSignatureAlgorithm()`. This PR is opened on master.

![image](https://user-images.githubusercontent.com/2013070/81967696-024fe180-9624-11ea-82a0-e8006cb489b4.png)

Reply:
thanks! changed.